src/components/ReplaceServerDetails.js

```diff
@@ -92,19 +150,26 @@ class ReplaceServerDetails extends Component {
       theProps.osInstallPassword = this.state.osInstallPassword; //TODO where to store
     }
 
-    //TODO the whole use available servers need further work
     // picked from available server, this is used to
     // update the available servers or manual servers list
     if(this.state.selectedServerId) {
       theProps.selectedServerId = this.state.selectedServerId;
-      // TODO if it is an existing server discovered or added, get its ID from availableServers
-      // this.props.availableServers.find(server => server.id === this.state.selectedServerId);
-      // was: data['uid'] = this.state.inputValue.get('uid');
+      // update internal uuid for UI purpose
+      let selServer =
+        this.props.availableServers.find(server => server.id === this.state.selectedServerId);
+      data['uid'] = selServer['uid'];
     }
     else {
       // user input new mac-addr and ilo info
-      // generate a new uid
-      data.uid = genUID();
+      // generate a new uid, treat it as manual added server
+      data['uid'] = genUID('manul');
```

Comment:
This dialog box should just be returning what the selectedServerId is, and the caller should be generatign a new ID if one is necessary. (BTW `manual` is misspelled here)

Reply:
will update